components/DeleteNodeDialog.tsx

```diff
@@ -21,19 +21,18 @@ export function DeleteNodeDialog(props: {
   const { projectId } = useProjectId();
   const dispatch = useStoreDispatch();
   const queryClient = useQueryClient();
-  const { choice, mainActivity } = NodeTypes;
+  const { choice, mainActivity, subActivity, waiting } = NodeTypes;
 
   const deleteMutation = useMutation(
     ({
       id,
       projectId,
-      type,
+      includeChildren,
     }: {
       id: string;
       projectId: string;
-      type: string;
-    }) =>
-      deleteVertice(id, projectId, type === mainActivity || type === choice),
+      includeChildren: boolean;
+    }) => deleteVertice(id, projectId, includeChildren),
```

Comment:
Id move the logic handling deleting of a node into a `useDeleteNodeMutation` to make this component only contain rendering logic and hooks.

Reply:
This is definitely something we will do for the next refactoring, but this isn't a priority now